### DISCUSSION TITLE
Would it make sense to merge `no-case-insensitive-enum-values-duplicates` and `unique-enum-value-names`?

### DISCUSSION THREAD

**Discussion by JustinTRoss** [graphql-hive/graphql-eslint#793] [Ideas]
Would it make sense to merge `no-case-insensitive-enum-values-duplicates` and `unique-enum-value-names`?

**Comment by dotansimha** [comment]
@B2o5T what do you think?

**Comment by dimaMachina** [comment]
@dotansimha `unique-enum-value-names` is `graphql-js`'s validation function, and `no-case-insensitive-enum-values-duplicates` is our internal rule. So merging them in one rule in `graphql-eslint` will be tricky. I don't want to change the behavior of external rules.

I found that we don't have a test for `no-case-insensitive-enum-values-duplicates`. I propose to let it be as it is and add test + fix report location for that rule.

**Reply by JustinTRoss** [reply]
Looking at the source for `no-case-insensitive-enum-values-duplicates`, it seems to already cover all use cases of `unique-enum-value-names`. The only difference is that the former always checks case-insensitive. If it instead took an option to toggle case-insensitivity in the check, there would effectively be no use for `unique-enum-value-names`, at which point this rule might be able to take over that namespace 😄 .

**Reply by JustinTRoss** [reply]
Looking through it now, for duplicate values of the same casing, they actually return a parse error instead of either of these rules triggering 😄. `Parsing error: Enum value "It.VALUE" can only be defined once`

**Reply by dimaMachina** [reply]
> Looking at the source for no-case-insensitive-enum-values-duplicates, it seems to already cover all use cases of unique-enum-value-names. 

Nice catch @JustinTRoss. So let's remove him from `recommended` and `all` config.

> they actually return a parse error instead of either of these rules triggering

that behavior because it's impossible to build schema with enum duplicates.

Also, I removed all rules with `'off'` option from `all` config after your suggestion https://github.com/dotansimha/graphql-eslint/pull/662#discussion_r753820534 and updated rules table 😄. https://github.com/B2o5T/graphql-eslint/tree/remove-unique-enum-value-names-from-recommended-config/docs

![image](https://user-images.githubusercontent.com/7361780/143135164-6ffedf8b-bc21-45d7-8cc9-cf42a8b1101c.png)

**Reply by JustinTRoss** [reply]
Loving the "Config" column. There was a point where I had all the rule pages opened in tabs to achieve this end 😅 .
